lib/src/over_react_test/prop_type_util.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+import 'dart:js';
+
+import 'package:react/react_client/react_interop.dart';
+
+/// Runs a provided callback and returns the logs that occur during the runtime
+/// of that function.
+///
+/// Can be used to capture logs, warnings, or errors as specified by setting
+/// the [configuration]. To set the [configuration], pass in the corresponding
+/// config class ([logConfig], [warnConfig], [errorConfig], [allConfig]).
+///
+/// If any errors are thrown during the callback, e.g. during a render that expects
+/// props that are not valid, the errors will be caught to allow the test to complete.
+///
+/// To handle asynchronous behavior, see [recordConsoleLogsAsync].
+List<String> recordConsoleLogs(Function() callback,
+    [ConsoleConfiguration configuration = allConfig]) {
+  final consoleLogs = <String>[];
+  final logTypeToCapture = configuration.logType == 'all'
+      ? ConsoleConfiguration.types
+      : [configuration.logType];
+  Map<String, JsFunction> consoleRefs = {};
+
+  PropTypes.resetWarningCache();
```

Comment:
Again, don't love that this in in here, because this shouldn't have anything to do with propTypes _technically_ ... :P

Reply:
might want to wrap this in a try catch JUST incase.